src/pages/index.js

```diff
@@ -66,40 +128,82 @@ profileEditButton.addEventListener("click", () => {
   editPopupForm.open();
 });
 
-// Function to render a card
+avatarEditButton.addEventListener("click", () => {
+  avatarEditPopup.open();
+});
+
+//to render a card
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleCardImageClick);
-  section.addItem(card.getCardElement());
+  const cardElement = new Card(
+    cardData,
+    "#card-template",
+    handleCardImageClick,
+    function handleDeleteClick() {
+      cardDeletePopup.setSubmitAction(() => {
+        cardDeletePopup.setLoading(true);
+        api
+          .removeCard(cardData._id)
+          .then((res) => {
+            console.log("here");
+            cardElement.removeCardElement(res._id);
+            cardDeletePopup.close();
+          })
+          .catch(console.error)
+          .finally(() => {
+            cardDeletePopup.setLoading(false);
+          });
+      });
+      cardDeletePopup.setEventListeners();
+      cardDeletePopup.open();
+    },
+    function handleLikeButtonClick() {
+      api
+        .changeLikeStatus(cardData._id, cardElement.isLiked)
+        .then((res) => cardElement.updateLikes(res.isLiked))
+        .catch(console.error);
+    }
+  );
+  // section.addItem(card.getCardElement());
+  return cardElement.getCardElement();
 }
 
-// Section for managing cards
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  cardsWrap
-);
-
-// Render initial cards
-section.renderItems();
-
 /* -------------------------------------------------------------------------- */
 /*                                  Handlers                                  */
 /* -------------------------------------------------------------------------- */
 
 // Event handler for profile edit form submission
 function handleEditProfileSubmit(formData) {
   const { nameInfo, jobInfo } = formData;
-  userInfo.setUserInfo(nameInfo, jobInfo);
-  editPopupForm.close();
+  editPopupForm.setLoading(true);
+  api
+    .setUserInfo(nameInfo, jobInfo)
+    .then(() => {
+      console.log("here");
+      userInfo.setUserInfo(nameInfo, jobInfo);
+      editPopupForm.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      editPopupForm.setLoading(false);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```